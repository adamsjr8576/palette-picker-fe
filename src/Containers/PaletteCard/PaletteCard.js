import { connect } from 'react-redux';
import React from 'react';
import './PaletteCard.scss';
import images from '../../images/images';
import { updatePaletteLocked } from '../../actions';

export const PaletteCard = ({ locked, hexCode, updatePaletteLocked }) => {
  const image = locked? images.locked : images.unlocked;
  return (
    <article className="palette-article" style={{backgroundColor: `${hexCode}`}}>
      <div className="palette-info-container">
        <section tabIndex="1" className="lock-button-section" role="update-palette=locked" onClick={() => updatePaletteLocked(hexCode)}>
          <img src={image} alt="lock icon" className="lock-button-image" />
        </section>
        <section className="hex-code-section">
          <p className="hex-p">{hexCode}</p>
        </section>
      </div>
    </article>
  );
};

export default PaletteCard;
