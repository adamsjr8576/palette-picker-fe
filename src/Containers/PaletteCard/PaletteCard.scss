.palette-article {
  width: 250px;
  height: 400px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-end;
}
.palette-info-container {
  width: 80%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  height: 100px;
}

.hex-code-section {
  width: 50%;
  min-width: 65px;
  border-radius: 20px;
  background-color: rgba(0, 0, 0, .4);
  padding: 5px;
  margin-bottom: 10px;
  color: white;
}

.lock-button-section {
  background-color: rgba(255, 255, 255, .6);
  border-radius: 50%;
  width: 55px;
  height: 55px;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
}

.lock-button-image {
  height: 35px;
  width: 35px;
}

.hex-p {
  margin: 0;
}
