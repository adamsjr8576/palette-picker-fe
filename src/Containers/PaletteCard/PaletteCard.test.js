import React from 'react';
import { PaletteCard } from './PaletteCard';
import { render, fireEvent } from '@testing-library/react';

describe('PaletteCard', () => {
  let mockProps, setup;

  test('should match the snapshot when locked is true', () => {
    mockProps = {
      locked: true,
      hexCode: '#11111',
      updatePaletteLocked: jest.fn()
    }

    setup = () => {
      const utils = render(<PaletteCard
        locked={mockProps.locked}
        hexCode={mockProps.hexCode}
        updatePaletteLocked={mockProps.updatePaletteLocked}
      />)
      return { utils };
    }

    const { utils } = setup();
    expect(utils).toMatchSnapshot();
  });

  test('should match the snapshot when locked is false', () => {
    mockProps = {
      locked: false,
      hexCode: '#11111',
      updatePaletteLocked: jest.fn()
    }

    setup = () => {
      const utils = render(<PaletteCard
        locked={mockProps.locked}
        hexCode={mockProps.hexCode}
        updatePaletteLocked={mockProps.updatePaletteLocked}
      />)
      return { utils };
    }

    const { utils } = setup();
    expect(utils).toMatchSnapshot();
  });

  test('should invoke updatePaletteLocked when button is clicked', () => {
    mockProps = {
      locked: false,
      hexCode: '#11111',
      updatePaletteLocked: jest.fn()
    }

    setup = () => {
      const utils = render(<PaletteCard
        locked={mockProps.locked}
        hexCode={mockProps.hexCode}
        updatePaletteLocked={mockProps.updatePaletteLocked}
      />);
      const btn = utils.getByRole('update-palette=locked')
      return {
        btn,
        ...utils
      };
    }

    const { btn } = setup();
    fireEvent.click(btn);
    expect(mockProps.updatePaletteLocked).toHaveBeenCalledWith(mockProps.hexCode);
  });
});
