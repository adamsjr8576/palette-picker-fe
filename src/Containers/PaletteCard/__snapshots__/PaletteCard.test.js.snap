// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteCard should match the snapshot when locked is false 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="palette-article"
      >
        <div
          class="palette-info-container"
        >
          <section
            class="lock-button-section"
            role="update-palette=locked"
            tabindex="1"
          >
            <img
              alt="lock icon"
              class="lock-button-image"
              src="unlock.png"
            />
          </section>
          <section
            class="hex-code-section"
          >
            <p
              class="hex-p"
            >
              #11111
            </p>
          </section>
        </div>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="palette-article"
    >
      <div
        class="palette-info-container"
      >
        <section
          class="lock-button-section"
          role="update-palette=locked"
          tabindex="1"
        >
          <img
            alt="lock icon"
            class="lock-button-image"
            src="unlock.png"
          />
        </section>
        <section
          class="hex-code-section"
        >
          <p
            class="hex-p"
          >
            #11111
          </p>
        </section>
      </div>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`PaletteCard should match the snapshot when locked is true 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="palette-article"
      >
        <div
          class="palette-info-container"
        >
          <section
            class="lock-button-section"
            role="update-palette=locked"
            tabindex="1"
          >
            <img
              alt="lock icon"
              class="lock-button-image"
              src="lock.png"
            />
          </section>
          <section
            class="hex-code-section"
          >
            <p
              class="hex-p"
            >
              #11111
            </p>
          </section>
        </div>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="palette-article"
    >
      <div
        class="palette-info-container"
      >
        <section
          class="lock-button-section"
          role="update-palette=locked"
          tabindex="1"
        >
          <img
            alt="lock icon"
            class="lock-button-image"
            src="lock.png"
          />
        </section>
        <section
          class="hex-code-section"
        >
          <p
            class="hex-p"
          >
            #11111
          </p>
        </section>
      </div>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
